src/click/core.py

```diff
@@ -1809,6 +1809,16 @@ def add_command(self, cmd: Command, name: t.Optional[str] = None) -> None:
         _check_multicommand(self, name, cmd, register=True)
         self.commands[name] = cmd
 
+    @t.overload
+    def command(self, __func: t.Callable[..., t.Any]) -> Command:
```

Comment:
```suggestion
    def command(self, __func: t.Callable[..., t.Any], *args: t.Any, **kwargs: t.Any) -> Command:
```
Looking at the code, it appears that any other arguments are passed to the command, so it should still accept additional arguments afterwards. Although probably not used much in practice...

Reply:
i.e. The code appears to explicitly support this:
`func = c.command(func, *args, **kwargs)`

Reply:
You cannot pass other arguments if you use it as a decorator without parenthesis. I've added assert statements requiring this be true. If you want to add other arguments, it should be used in the traditional way as a factory. And mypy certainly should not encourage that even if was supported at runtime!